applicationinsights-react-js/src/ReactPlugin.ts

```diff
@@ -5,16 +5,24 @@
 
 import dynamicProto from "@microsoft/dynamicproto-js";
 import {
+    AnalyticsPluginIdentifier,
     IAppInsights, IConfig, IEventTelemetry, IExceptionTelemetry, IMetricTelemetry, IPageViewTelemetry, ITraceTelemetry
 } from "@microsoft/applicationinsights-common";
 import {
     BaseTelemetryPlugin, IAppInsightsCore, IConfiguration, ICookieMgr, ICustomProperties, IPlugin, IProcessTelemetryContext,
     IProcessTelemetryUnloadContext, ITelemetryItem, ITelemetryPlugin, ITelemetryPluginChain, ITelemetryUnloadState, _eInternalMessageId,
-    _throwInternal, arrForEach, eLoggingSeverity, isFunction, objDefineAccessors, proxyFunctions, safeGetCookieMgr
+    _throwInternal, arrForEach, eLoggingSeverity, isFunction, proxyFunctions, safeGetCookieMgr, IConfigDefaults, onConfigChange, objDefineAccessors
 } from "@microsoft/applicationinsights-core-js";
+import {objDeepFreeze} from "@nevware21/ts-utils";
+
 import { History, Location, Update } from "history";
 
 import { IReactExtensionConfig } from './Interfaces/IReactExtensionConfig';
+const defaultReactExtensionConfig: IConfigDefaults<IReactExtensionConfig> = objDeepFreeze({
+    blkVal: true,
```

Comment:
Not exactly, this is defining a "property" called blkVal.

You need to either use the helper function of the "default" for the history need to contain the value

```ts
const defaultReactExtensionConfig: IConfigDefaults<IReactExtensionConfig> = objDeepFreeze({
    history: { blkVal: true, v: undefined }
});
```

This is because the "value" of the default can either be the "value" you want to use as the default (for simple properties) or an `IConfigDefaults` instance, which contains additional attributes to associate with the property (`history` in this case) and any value assigned to this property.

Reply:
Just for double check. if user change history object, the prev_history would not equal to the current history, and the onConfigChange method would be called. If they change history.path, the prev_history.path would also not equal to current_history.path, but the onConfigChange would not be called.

Reply:
>  If they change history.path, the prev_history.path would also not equal to current_history.path, but the onConfigChange would not be called.

Correct, and that is fine, we are only adding hooks for when they change the "histroy" obejct that they are using.
During a SPA application (which react is used for) the `path` value of the `history` WILL (most often) change this code does not need (or want) to track this change (that is what the AnalyticsPlugin) is doing.